src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Commit.cs

```diff
@@ -120,15 +121,17 @@ private void Commit(CompletionItem item, TextChange textChange, Model model, cha
                 }
             }
 
-            if (item.ShouldFormatOnCommit)
+            var document = this.SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            var formattingService = document.GetLanguageService<IEditorFormattingService>();
+            if (formattingService != null && 
+                (item.ShouldFormatOnCommit || (commitChar != null && formattingService.SupportsFormattingOnTypedCharacter(document, commitChar.GetValueOrDefault()))))
             {
                 // Formatting the completion item affected span is done as a separate transaction because this gives the user
                 // the flexibility to undo the formatting but retain the changes associated with the completion item
                 using (var formattingTransaction = _undoHistoryRegistry.GetHistory(this.TextView.TextBuffer).CreateTransaction(EditorFeaturesResources.IntellisenseCommitFormatting))
```

Comment:
If you look at the top of this file (thanks github) you'll see that we use a different string for IntelliSense (EditorFeaturesResources.IntelliSense).